deepspeed/comm/comm.py

```diff
@@ -0,0 +1,541 @@
+"""
+    Copyright 2021 The Microsoft DeepSpeed Team
+
+    DeepSpeed Communication Package: deepspeed.comm
+
+    deepspeed.comm
+        -- import and use deepspeeed.ops.comm
+        -- use torch.distributed directly if both this package and torch.distributed use the same NCCL version
+        -- use custom collectives
+            -- can either use torch.dist or ds.ops.comm?
+
+        Note: the old 1-bit compressed allreduce variants that resided in deepspeed.runtime.comm will be moved here as well.
+
+    deepspeed.comm API
+        -- must be kept fully compatible (same signatures) as torch.dist API to ensure backward/cross-framework compatibility.
+        -- e.g. if a client code used
+            import deepspeed.comm as dist
+
+            instead of
+            import torch.distributed as dist
+
+            The code should work without breaking any of the public torch.distributed functionality
+
+    Future:
+        -- deepspeed groups API should be brought into ds.comm
+"""
+
+from enum import Enum
+import torch
+import os
+import torch
+
+from ..constants import TORCH_DISTRIBUTED_DEFAULT_PORT, default_pg_timeout
+
+
+class ReduceOp(Enum):
+    SUM = 0
+    PRODUCT = 1
+    MIN = 2
+    MAX = 3
+    BAND = 4
+    BOR = 5
+    BXOR = 6
+    AVG = 7
+    UNUSED = 8
+
+
+from deepspeed.comm.backend import Backend
+from deepspeed.comm.torch import TorchBackend
+
+import deepspeed.utils as utils
+from datetime import timedelta
+
+# Current deepspeed.comm backend (cdb) global object for simple access by client code
+use_ds_backend = False
+cdb = None
```

Comment:
What does cdb mean?

Reply:
deepspeed comm backend => dcb?

Reply:
It stands for "current deepspeed.comm backend": https://github.com/microsoft/DeepSpeed/blob/staging-comms-next/deepspeed/comm/comm.py#L54

I'm OK with a cdb => dcb refactor if @awan-10 is. I agree that it's a more natural var name.

Reply:
Either way is fine with me. It was not obvious to me.